back-nest/src/sсrapers/scraper-telemart.service.ts

```diff
@@ -1,149 +1,112 @@
 
-        const description =
-          element
-            .querySelector('.product-card__description')
-            ?.textContent?.trim() || 'No description available';
+      this.logger.log('Extracting product data...');
+      const rawProducts: ScrapedProduct[] = await page.evaluate(() => {
+        const items = Array.from(
+          document.querySelectorAll('.product-item__inner'),
+        );
 
-        const price = Number(element.getAttribute('data-price')) || 0;
+        return items.map<ScrapedProduct>((element) => {
+          const title =
+            element
+              .querySelector('.product-item__title a')
+              ?.textContent?.trim() || 'No title available';
 
-        const specifications: { [key: string]: string } = {};
+          const subtitle =
+            element
+              .querySelector('.product-item__title a')
+              ?.getAttribute('href') || 'No subtitle available';
 
-        element
-          .querySelectorAll('.product-short-char__item')
-          .forEach((specElement) => {
-            const label =
-              specElement
-                .querySelector('.product-short-char__item__label')
-                ?.textContent?.trim() || 'Unknown';
+          const description =
+            element
+              .querySelector('.product-card__description')
+              ?.textContent?.trim() || 'No description available';
 
-            specifications[label] =
-              specElement
-                .querySelector('.product-short-char__item__value')
-                ?.textContent?.trim() || 'Unknown';
-          });
+          const price = Number(element.getAttribute('data-price')) || 0;
 
-        const type = element.getAttribute('data-prod-type') || 'Unknown type';
+          const specifications: { [key: string]: string } = {};
+          element
+            .querySelectorAll('.product-short-char__item')
+            .forEach((specElement) => {
+              const label =
+                specElement
+                  .querySelector('.product-short-char__item__label')
+                  ?.textContent?.trim() || 'Unknown';
+
+              specifications[label] =
+                specElement
+                  .querySelector('.product-short-char__item__value')
+                  ?.textContent?.trim() || 'Unknown';
+            });
+
+          const type = element.getAttribute('data-prod-type') || 'Unknown type';
+
+          const imageElement = element.querySelector(
+            '.swiper-slide.swiper-slide-active img',
+          );
+
+          const profileImage = imageElement
+            ? imageElement.getAttribute('src')
+            : null;
+
+          return {
+            title,
+            subtitle,
+            description,
+            price,
+            specifications: JSON.stringify(specifications),
+            type,
+            profileImage,
+            source: 'TELEMART',
```

Comment:
Move source into `constants` file. Do the same for rozetka. Also replace all string occurencies in your code with constant

```ts
const sources = {
  Telemart: 'TELEMART',
  Rozetka: 'ROZETKA',
}
```

```suggestion
            source: Sources.Telemart,
```